server/scripts/dbNode.ts

```diff
@@ -153,78 +153,81 @@ async function mainProcess(
+      if (
+        tokens.length > 0 &&
+        (!listeners[`erc20_${url}`] ||
+          (listeners[`erc20_${url}`] &&
+            !_.isEqual(
+              erc20TokenAddresses,
+              listeners[`erc20_${url}`].options.tokenAddresses
+            )))
+      ) {
+        // clear the listener if it already exists and the tokens have changed
+        if (listeners[`erc20_${url}`]) {
+          listeners[`erc20_${url}`].unsubscribe();
+          delete listeners[`erc20_${url}`];
+        }
 
-      // if listener has started at this point then subscribe
-      if (listeners['erc20']) {
+        // start a listener
+        log.info(`Starting listener for ${erc20TokenNames}...`);
         try {
-          // subscribe to the chain to begin listening for events
-          await listeners['erc20'].subscribe();
+          listeners[`erc20_${url}`] = await createListener(
+            'erc20',
+            SupportedNetwork.ERC20,
+            {
+              url,
+              tokenAddresses: erc20TokenAddresses,
+              tokenNames: erc20TokenNames,
+              verbose: false,
+            }
+          );
+
+          // add the rabbitmq handler for this chain
+          listeners[`erc20_${url}`].eventHandlers['rabbitmq'] = { handler: producer };
+          listeners[`erc20_${url}`].eventHandlers['logger'] = { handler: erc20Logger };
         } catch (error) {
-          await handleFatalError(error, pool, 'erc20', 'listener-subscribe');
+          delete listeners[`erc20_${url}`];
+          await handleFatalError(error, pool, `erc20_${url}`, 'listener-startup');
+        }
+
+        // if listener has started at this point then subscribe
+        if (listeners[`erc20_${url}`]) {
+          try {
+            // subscribe to the chain to begin listening for events
+            await listeners[`erc20_${url}`].subscribe();
+          } catch (error) {
+            await handleFatalError(error, pool, `erc20_${url}`, 'listener-subscribe');
+          }
         }
+      } else if (listeners[`erc20_${url}`] && tokens.length === 0) {
+        // delete the listener if there are no tokens to listen to
+        log.info(`[erc20_${url}]: Deleting erc20 listener...`);
+        listeners[`erc20_${url}`].unsubscribe();
+        delete listeners[`erc20_${url}`];
       }
-    } else if (listeners['erc20'] && erc20Tokens.length === 0) {
-      // delete the listener if there are no tokens to listen to
-      log.info('[erc20]: Deleting erc20 listener...');
-      listeners['erc20'].unsubscribe();
-      delete listeners['erc20'];
+    } else {
+      log.fatal(
+        `[erc20_${url}]: There are outstanding errors that need to be resolved before creating a new erc20 listener!`
+      );
     }
-  } else {
-    log.fatal(
-      '[erc20]: There are outstanding errors that need to be resolved before creating a new erc20 listener!'
-    );
   }
 
   // remove erc20 tokens from myChainData
```

Comment:
The code below needs to be changed so that the new ERC20 Event Listeners are not deleted.
Line 242 could be something like: `if (!myChains.includes(chain) && !chain.includes('erc20')) {`

Reply:
Good catch! Fixed in most recent commit. How would I go about testing this code?

Reply:
Set `has_chain_events_listener` to true in `Chains` for a few erc20 tokens (usd-coin, dai-stablecoin are good ones). Then also set `ce_verbose` to true for those tokens in `ChainNodes`. Then run 
```
yarn build-scripts && NUM_WORKERS=1 WORKER_NUM=0 node build/server/scripts/dbNode.js
```
This will run a single dbNode instance with all the chains/tokens on it. Maybe try to include a few different tokens that use different URLs so you can make sure its creating different listeners for different URLs correctly.

I just created a small PR that further improves logging in `dbNode.ts` that if included before this PR would make testing this PR easier as the log prefix could show up with the URL (with very slight changes) thus confirming that there are indeed different listeners being created.